src/python/gem5/resources/downloader.py

```diff
@@ -253,6 +254,16 @@ def get_resource(
             gem5_version=gem5_version,
         )
 
+        if (
+            "url" in resource_json
+            and resource_json["url"]
+            and (urlparse(resource_json["url"]).scheme == "file")
+        ):
+            local_path = urlparse(resource_json["url"]).path
+            if os.path.exists(local_path):
+                shutil.copy(local_path, to_path)
+                return
+
```

Comment:
I'm not sure this covers all of the specification here. I think i want to see tests I'm afraid. Could you do the following?:

1. Move this into a private function `def _file_uri_to_path(uri: str) -> Optional[Path]`. This function will return a path if the uri is a file, otherwise none.
2. Update `get_resource` to use this function.
3. Using the pyunit test framework, add unittests for this new function. I would look over https://en.wikipedia.org/wiki/File_URI_scheme to create test input data and expected outputs.

As a small example of input/output pairs to test:

|Input|Output|
|---|---|
|https://gem5.org/bla|None|
|file:///home/bbruce/binary|/home/bbruce/binary|
|file:/home/bbruce/binary|/home/bbruce/binary|
|file://localhost/home/bbruce/binary|/home/bbruce/binary|
|file:////localhost/home/bbruce/binary|/home/bbruce/binary|
|file://other-host/home/bbruce/binary| Exception("Can only obtain file from localhost") | 

Add more so we're covering every case here. Don't just copy and paste, have a think about what kind of behavior we want to verify here.

Reply:
Alright! Seems fair to me, I will also make sure to add in-line documentation to the new function as well.